src/client/Inode.h

```diff
@@ -193,6 +196,16 @@ struct Inode : RefCountedObject {
 
   // use i_flags as 1 << 14 will overlap with other mode bits.
   bool is_encrypted() const { return (i_flags & S_ENCRYPTED) == S_ENCRYPTED; }
+  // this function sets S_ENCRYPTED bit in i_flag
+  // is called when the is_fscrypt_enabled
+  void set_is_encrypted_flag() {
+    bool en = is_fscrypt_enabled();
+    if (en) {
+      i_flags |= S_ENCRYPTED;
+    } else {
+      i_flags &= (~S_ENCRYPTED);
```

Comment:
Same as above:) For better semantics and defensive code.

Reply:
I see your reasoning on this. I'm concerned that if there's "garbage" in it, we should let it show instead of hiding the issue.

I'm looking at linux kernel ioctl.c and during 
`fileattr_fill_flags|fileattr_fill_xflags`

it sets value and doesn't worry about clearing.

Reply:
It makes sense for me now.I think that best will be to assert(false) if `is_fscrypt_enabled` is false but the S_ENCRYPTED bit is set. It wil let us catch the problem, instead of cleaning it.

Reply:
I like that, let's proceed with that change.